Source/DafnyDriver/Commands/TestCommand.cs

```diff
@@ -0,0 +1,30 @@
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Linq;
+
+namespace Microsoft.Dafny; 
+
+public class TestCommand : ICommandSpec {
+  public IEnumerable<IOptionSpec> Options =>
+    new IOptionSpec[] {
+      TargetOption.Instance,
+      NoVerifyOption.Instance,
+      EnforceDeterminismOption.Instance,
+      VerificationTimeLimitOption.Instance,
+    }.Concat(ICommandSpec.CommonOptions);
+
+  public Command Create() {
+    var result = new Command("test", "(Experimental) Execute every method in the program that's annotated with the {:test} attribute.");
+    result.AddArgument(ICommandSpec.FilesArgument);
+    return result;
+  }
+
+  public void PostProcess(DafnyOptions dafnyOptions, Options options, InvocationContext context) {
+    dafnyOptions.Compile = true;
+    dafnyOptions.RunAfterCompile = true;
+    dafnyOptions.RunAllTests = true;
+    dafnyOptions.ForceCompile = NoVerifyOption.Instance.Get(options);
+    dafnyOptions.CompileVerbose = false;
+  }
+}
```

Comment:
Does this return the right error code when a test fails?

Reply:
I don't think it does. The behavior is determined by the code in `/Users/rwillems/Documents/SourceCode/dafny/Source/DafnyCore/Resolver/RunAllTestsMainMethod.cs`. I'm guessing Dafny currently doesn't have a way of letting its programs return a non-zero exit code. Letting `dafny test` return the right exit code seems like a significant amount of work, delaying that seems worth the breakage of having to modify the `dafny test` exit code later on.

Reply:
I think we recently merged a fix for this in `/runAllTests` (relevant issue is https://github.com/dafny-lang/dafny/issues/3043); let's just check that the fix also works with the new `dafny test`.

Reply:
Tests now check both the zero and non-zero exit code scenario.